modules/vaos/app/controllers/vaos/v2/referrals_controller.rb

```diff
@@ -116,6 +104,49 @@ def sanitize_log_value(value)
 
         value.to_s.gsub(/\s+/, '_')
       end
+
+      # Logs referral provider metrics and errors for missing provider IDs
+      # @param response [Ccra::ReferralDetail] the referral response object
+      def log_referral_provider_metrics(response)
+        # Check original values before sanitization for logging
+        original_referring_id = response&.referring_facility_code
+        original_referral_id = response&.provider_npi
+
+        # Sanitize for metrics
+        referring_provider_id = sanitize_log_value(original_referring_id)
+        referral_provider_id = sanitize_log_value(original_referral_id)
+
+        StatsD.increment(REFERRAL_DETAIL_VIEW_METRIC, tags: [
+                           'service:community_care_appointments',
+                           "referring_provider_id:#{referring_provider_id}",
+                           "referral_provider_id:#{referral_provider_id}"
+                         ])
+
+        log_missing_provider_ids(original_referring_id, original_referral_id, response&.station_id)
+      end
+
+      # Logs specific errors when provider IDs are missing
+      # @param referring_provider_id [String] the original referring provider ID
+      # @param referral_provider_id [String] the original referral provider ID
+      # @param station_id [String] the station ID of the referral
+      def log_missing_provider_ids(referring_provider_id, referral_provider_id, station_id)
+        missing_fields = []
+        missing_fields << REFERRING_PROVIDER_ID if referring_provider_id.blank?
+        missing_fields << REFERRAL_PROVIDER_ID if referral_provider_id.blank?
+
+        return if missing_fields.empty?
+
+        description = if missing_fields.size == 1
+                        "#{missing_fields.first} is"
+                      else
+                        BOTH_PROVIDER_IDS
+                      end
+
+        sanitized_station_id = sanitize_log_value(station_id)
+
+        Rails.logger.error("Community Care Appointments: Referral detail view: #{description} blank for user: " \
```

Comment:
Nit: This will work but creating queries will be more difficult. I think if you do something like `Rails.logger.warn("Community Care Appointments: Referral detail view: Missing provider data", {missing_data:["referring_provider_id", "referral_provider_id"], station_id: sanitized_station_id, user_uuid: current_user.uuid})` then we would get key value pairs in the log payload.